### PR TITLE
Deletes all deprecated API

### LINKED ISSUE #466: Delete deprecated enterState method
It has a pathological debug name function, and has been deprecated for a long time. Make it go away.

### PR DESCRIPTION
**Pull request by rjrjr**
closes #466

### CODE REVIEW

File: samples/hello-terminal/terminal-workflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/KeyStroke.kt [hunk 1]
```diff
@@ -46,16 +49,18 @@ data class KeyStroke(
...
 internal fun InputProvider.listenForKeyStrokesOn(
   scope: CoroutineScope
-): BroadcastChannel<KeyStroke> = scope.broadcast {
-  while (isActive) {
-    val keyStroke = readInput()
-    if (keyStroke.keyType === EOF) {
-      // EOF indicates the terminal input was closed, and we won't receive any more input, so
-      // close the channel instead of sending the raw event down.
-      return@broadcast
+): SharedFlow<KeyStroke> {
```
Comment (rjrjr) [inline samples/hello-terminal/terminal-workflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/KeyStroke.kt:52]: Does this really need to be a `SharedFlow`? I was just mechanically following the conversion steps from the `BroadcastChannel` deprecation docs.

**Review by zach-klippenstein — APPROVED** [review]
LG, but discussed offline holding merge until alpha21 is out.

**Comment by zach-klippenstein** [comment]
Released, merge away!

**Comment by Kernald** [comment]
Note that this breaks most of the Android samples and, maybe more importantly, the tutorial.

**Pull request merged**